File: Codeforces/round_794/task_A.scala
// https://codeforces.com/contest/1686/problem/0
//package round_794

import scala.io.StdIn

object task_A extends App {
    def solving(a: Array[Int], n: Int): String = {
        val avg = a.sum.toFloat / n
        var i = 0
        while (i < n && a(i) != avg) {
            i += 1
        }
        if (i == n) {
            "NO"
        }
        else {
            "YES"
        }
    }

    val t = StdIn.readLine().toInt
    var sb = new StringBuilder()
    for (i <- 0 until(t)) {
        val n = StdIn.readLine().toInt
        val a: Array[Int] = StdIn.readLine().split(" ").map(x => x.toInt)
        sb.append(solving(a, n)).append("\n")
    }
    print(sb)

}

File: coding/Codeforces/ed_round_131/task_A.py
# https://codeforces.com/contest/1701/problem/A

def solving(arr):
    ans = 0
    sum_arr = sum([sum(arr[i]) for i in range(2)])
    if sum_arr == 0:
        ans = 0
    elif 1 <= sum_arr <= 3:
        ans = 1
    else:
        ans = 2
    return ans

def main():
    t = int(input())
    answer = []
    for _ in range(t):
        arr = [list(map(int, input().split())) for _ in range(2)]
        answer.append(solving(arr))
    answer = "\n".join(map(str, answer))
    print(answer)

if __name__ == "__main__":
    main()